cibyl/sources/zuul/utils/variants/hierarchy.py

```diff
@@ -219,12 +222,21 @@ def __next__(self) -> Variant:
             self._step = next_step
             return self._step
 
-        def _get_next_step(self) -> Variant:
+        def _get_next_step(self) -> Optional[Variant]:
             # Make the start variant be part of the iteration
             if self._step is None:
                 return self._start
 
-            return self.tools.variants.find().parent_of(self._step)
+            try:
+                return self.tools.variants.find().parent_of(self._step)
+            except SearchError as ex:
```

Comment:
No related to this change at all, but `SearchError` should inherit from `SourceException` to catch it properly when running queries.